app/views/govuk_publishing_components/components/docs/layout_for_public.yml

```diff
@@ -123,3 +123,11 @@ examples:
           cookie_preferences:
             text: How GOV.UK accounts use cookies
             href: https://www.gov.uk/government/publications/govuk-accounts-trial-full-privacy-notice-and-accessibility-statement
+  with_custom_layout:
+    description: Yields a custom layout for the content.
+    data:
+      custom_layout: true
+      block: |
+        <main id="custom-layout">
+          <h1>This is a custom layout</h1>
+        </main>
```

Comment:
I imagine we should also have an option that demonstrates custom header, though I know it's a bit tricky to know how to do so.

This is what I whipped up:

```
  with_custom_header:
    description: Allows the header to be replaced with HTML injected by the calling application
    embed: |
      <% content_for(:custom_header) do %>
        <header>I'm a custom header</header>
      <% end %>
      <%= component %>
```

Reply:
Cool thanks Kevin, I’ve made that change now 👍 I had to use `<%= render "govuk_publishing_components/components/layout_for_public", {} %>` instead of `<%= component %>` otherwise the HTML preview kept rendering the default header for some reason